roles/openshift_health_checker/library/search_journalctl.py

```diff
@@ -0,0 +1,163 @@
+            break
+
+        try:
+            obj = json.loads(line.rstrip())
+
+            # don't need to look past the most recent service restart
+            if start_regexp.match(obj["MESSAGE"]):
+                break
+
+            if stamp_too_old(obj["__REALTIME_TIMESTAMP"], timestamp_limit_seconds):
+                break
+
+            if regexp.match(obj["MESSAGE"]):
+                matched = line.rstrip()
+                break
+
+        except ValueError as err:
+            return None, str(err)
+
+    return matched, None
+
+
+def ensure_matcher_fields(matcher):
+    """Ensures that expected fields exist on a user-defined matcher dict"""
+    required_fields = [
+        "start_regexp",
+        "regexp",
+        "unit",
+    ]
+    missing_fields = set(required_fields) - set([field for field in matcher])
+    if len(missing_fields):
+        raise InvalidMatcherException("Matcher missing required fields", list(missing_fields))
+
+
+def _get_log_matches(matchers, log_count_limit, timestamp_limit_seconds):
+    matched_regexp = []
+    errors = []
+
+    for matcher in matchers:
+        try:
+            ensure_matcher_fields(matcher)
+        except InvalidMatcherException as err:
+            errors.append(str(err))
+            continue
+
+        log_input, error = _get_log_input(matcher)
+        if error:
+            errors.append(error)
+            continue
+
+        try:
+            matched, error = find_matches(log_input, matcher, log_count_limit, timestamp_limit_seconds)
+            if matched:
+                matched_regexp.append(matcher["regexp"])
+                continue
+        except InvalidMatcherRegexpException as err:
+            error = str(err)
+
+        if error:
+            errors.append(error)
+
+    return matched_regexp, errors
+
+
+def _get_log_input(matcher):
+    try:
+        cmd_output = subprocess.Popen(list([
+            '/bin/journalctl',
+            '-ru', matcher["unit"],
+            '--output', 'json',
+        ]), stdout=subprocess.PIPE)
+
+        return iter(cmd_output.stdout.readline, ''), None
+
+    except subprocess.CalledProcessError as exc:
+        err = '[rc {}] {}\n{}'.format(exc.returncode, ' '.join(exc.cmd), exc.output)
+        return None, str(err)
+
+    except OSError as exc:
+        return None, str(exc)
+
+
+class InvalidMatcherException(Exception):
+    """Base exception class for missing or invalid matcher fields"""
+    def __init__(self, message, missing_fields=None):
+        Exception.__init__(self, message)
+        self.missing_fields = missing_fields
+
+
+def main():
+    """Scans a given list of matchers for journalctl messages matching given patterns"""
+    module = AnsibleModule(
+        argument_spec=dict(
+            log_count_limit=dict(type="int", default=500),
+            log_matchers=dict(type="list", required=True),
+        ),
+    )
+
+    timestamp_limit_seconds = time() - 60 * 60  # 1 hour
```

Comment:
Suggestion: use a single time unit and avoid conversions.

We pass this `timestamp_limit_seconds` around through several method calls, and then in `stamp_too_old` we receive time arguments in two different units and need to convert before comparing...